### DISCUSSION TITLE
Dead-end error with Infinity datasource

### DISCUSSION THREAD

**Discussion by toddrimes** [grafana/grafana-infinity-datasource#733] [Q&A]
We're using the Infinity Datasource plugin from Grafana Labs and we are getting a 'black-box' error from the plugin. How can we find out exactly what the issue is?

![image](https://github.com/grafana/grafana-infinity-datasource/assets/363950/820399bb-52d8-46f2-ba30-b94849716024)

**Comment by yesoreyeram** [comment]
Can you check grafana logs? Also let us know the steps to reproduce

**Reply by toddrimes** [reply]
The error seems to be happening on the remote (I think?) Infinity server which proxies the URL to the actual internal data source, which itself is also proxied. Does the remote Infinity server not log errors?

**Reply by toddrimes** [reply]
It seems to work _intermittently_ if I reduce the size of the JSON payload by reducing the number of records requested.  Is there a limit in record count or total size that Infinity can handle?

**Reply by yesoreyeram** [reply]
> It seems to work intermittently if I reduce the size of the JSON payload by reducing the number of records requested. Is there a limit in record count or total size that Infinity can handle?

Infinity doesn't have any limitations on record count. I suspect your underlying API is intermittently times out or doesn't handle the load properly causing the issue.
